flake.nix

```diff
@@ -6,46 +6,69 @@
       url = "github:oxalica/rust-overlay";
       inputs.nixpkgs.follows = "nixpkgs";
     };
-    naersk.url = "github:nix-community/naersk";
   };
 
-  outputs = { self, nixpkgs, rust-overlay, naersk, flake-utils }:
+  outputs = { self, nixpkgs, rust-overlay, flake-utils }:
     let systems = [ "x86_64-linux" "aarch64-linux" ];
     in flake-utils.lib.eachSystem systems (system:
       let
         overlays = [ (import rust-overlay) ];
         pkgs = import nixpkgs { inherit system overlays; };
         lib = pkgs.lib;
 
-        naersk' = pkgs.callPackage naersk {
-          cargo = pkgs.rust-bin.stable.latest.default;
-          rustc = pkgs.rust-bin.stable.latest.default;
-        };
+        cargoToml = builtins.fromTOML (builtins.readFile ./Cargo.toml);
+        cargoPackageVersion = cargoToml.package.version;
 
-        buildXwaylandSatellite = { withSystemd ? false }: naersk'.buildPackage {
-          src = ./.;
+        commitHash = self.shortRev or self.dirtyShortRev or "unknown";
 
-          nativeBuildInputs = with pkgs; [
-            rustPlatform.bindgenHook
-            rust-bin.stable.latest.default
-            pkg-config
+        version = "${cargoPackageVersion}-unstable-${commitHash}";
 
-            xcb-util-cursor
-            xorg.libxcb
+        buildXwaylandSatellite = { withSystemd ? true, ... } @ args:
```

Comment:
Indeed, I had only made the `withSystemd` option overridable because that was the primary goal of the refactoring without considering the dependencies. This is now addressed in 4c74de261f22165b1ea7002380ed55530b50a8b4.